### PR TITLE
amam/add_dropdown

### PR DESCRIPTION
**Pull request by mustofa-binary**
# Description

add keyboard friendly dropdown

Changes:

-   use KEY_CODES
-   add dropdown for language switcher

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Refactor code
-   [ ] Update translation text
-   [ ] Breaking change
-   [ ] Dependency update
-   [ ] This change requires a documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/elements/dropdown.js [hunk 1]
```diff
@@ -1,11 +1,243 @@
...
+    overflow: hidden;
+    padding: 0 1.6rem;
+    text-overflow: ellipsis;
+    height: 100%;
+    font-size: var(--text-size-xs);
+    display: flex;
+    align-items: center;
+
+    &:focus {
+        outline: none;
```
Comment (oskar-binary) [inline src/components/elements/dropdown.js]: this is as per the styleguide?

File: src/components/elements/dropdown.js [hunk 2]
```diff
@@ -1,11 +1,243 @@
...
 `
 
-export default StyledSelect
+const Dropdown = ({ default_option, onChange, option_list }) => {
+    const SPACEBAR_KEY_CODE = [0, 32]
+    const ENTER_KEY_CODE = 13
+    const TAB_KEY_CODE = 9
+    const DOWN_ARROW_KEY_CODE = 40
+    const UP_ARROW_KEY_CODE = 38
+    const ESCAPE_KEY_CODE = 27
```
Comment (oskar-binary) [inline src/components/elements/dropdown.js]: let's move the keycodes to a sharable object in common

File: src/components/elements/dropdown.js [hunk 3]
```diff
@@ -1,11 +1,243 @@
...
+                role="button"
+                id="selected_dropdown"
+                tabIndex="0"
+                onClick={toggleListVisibility}
+                onKeyDown={toggleListVisibility}
+            >
+                {selected_option}
+            </DropdownSelected>
+            <Arrow expanded={is_open} />
+            <ListContainer aria-expanded={`${is_open ? 'true' : 'false'}`} role="list">
```
Comment (oskar-binary) [inline src/components/elements/dropdown.js]: aria-expanded nice 👍

File: src/components/elements/dropdown.js [hunk 4]
```diff
@@ -1,11 +1,243 @@
...
+
+    const toggleListVisibility = e => {
+        e.preventDefault()
+        const open_dropdown =
+            SPACEBAR_KEY_CODE.includes(e.keyCode) ||
+            e.keyCode === ENTER_KEY_CODE ||
+            e.keyCode === TAB_KEY_CODE
+        Array.from(nodes.values())
+            .filter(node => node !== null)
+            .forEach(node => getPosition(node))
```
Comment (oskar-binary) [inline src/components/elements/dropdown.js]: what's going on here?

File: src/components/elements/dropdown.js (hunk 1 above)
Comment (mustofa-binary) [reply src/components/elements/dropdown.js]: by default, there will be blue2 outline if got focus. maybe can make it the same as active which is green border

File: src/components/elements/dropdown.js (hunk 4 above)
Comment (mustofa-binary) [reply src/components/elements/dropdown.js]: let me put a comment on it 👍

File: src/components/elements/dropdown.js (hunk 1 above)
Comment (oskar-binary) [reply src/components/elements/dropdown.js]: would be nice to have it in the official styleguide https://a11yproject.com/posts/never-remove-css-outlines/

File: src/common/keycodes.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const Keycodes = {
+    space: 32,
+    enter: 13,
+    tab: 9,
+    down_arrow: 40,
+    up_arrow: 38,
+    escape: 27,
```
Comment (oskar-binary) [inline src/common/keycodes.js]: make all caps?

**Pull request merged**